DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/non_std_convariance.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef __NON_STD_COVARIANCE_MATRIX_HPP__
+#define __NON_STD_COVARIANCE_MATRIX_HPP__
+
+namespace fpga_linalg {
+// This functor computes the columns x columns covariance matrix of a rows x
+// columns input matrix A
+
+// The deign compute T tah is is transpose(A)*A 
+
+template <typename T,          // The datatype for the computation
+          unsigned rows,       // Number of rows in the A matrices
+          unsigned columns,    // Number of columns in the A matrices
+          unsigned pipe_size,  // Number of elements read/write per pipe
+                               // operation, the matrix is received through the
+                               // pipe
+                               // by blocks of size columns*columns.
+          typename InputPipe,  // A matrix input pipe, receive pipe_size
+                               // elements from the pipe with each read
+          typename OutputPipe  // Q matrix output pipe, send pipe_size
+                               // elements to the pipe with each write
```

Comment:
```suggestion
          typename OutputPipe  // T matrix output pipe, send pipe_size
                               // elements to the pipe with each write
```
I assume that calling this Q matrix is a typo